docs/Modders/Mod-file-structure/2-Civilization-related-JSON-files.md

```diff
@@ -6,204 +6,270 @@
-| requiredResource | String | defaults to none | The resource that is consumed when building this building. Must be in [TileResources.json](3-Map-related-JSON-files.md#tileresourcesjson) |
-| requiredNearbyImprovedResources | List of Strings | defaults to none | The building can only be built if any of the resources in this list are within the borders of this city and have been improved. Each resource must be in [TileResources.json](3-Map-related-JSON-files.md#tileresourcesjson) |
-| replaces | String | defaults to none | The name of a building that should be replaced by this building. Must be in [Buildings.json](#buildingsjson) |
-| uniqueTo | String | defaults to none | If supplied, only the nation with this name can build this building. Must be in [Nations.json](#nationsjson) |
-| xpForNewUnits | Integer | defaults to 0 | XP granted automatically to units built in this city |
-| cityStrength | Integer | defaults to 0 | Strength bonus the city in which this building is built receives |
-| cityHealth | Integer | defaults to 0 | Health bonus the city in which this building is built receives |
-| hurryCostModifier | Integer | defaults to 0 | When this building is bought using gold or faith, the price is increased by this much percent |
-| quote | String | defaults to none | If this building is a (national) wonder, this string will be shown on the completion popup |
-| uniques | List of Strings | defaults to none | List of unique abilities this building has. Most of these can be found [here](../Modders/uniques) |
-| replacementTextForUniques | String | defaults to none | If provided, this string will be shown instead of all of the uniques |
-| percentStatBonus | Object | defaults to none | Percentual bonus for stats provided by the building. Valid keys are the names of stats (production, gold, science, etc.), valid values are Integers (≥0) |
-| greatPersonPoints | Object | defaults to none | How many great person points for each type will be generated per turn. Valid keys are the names of great people (Great Scientist, Great Engineer, etc. .), valid values are Integers (≥0) |
-| specialistSlots | Object | defaults to none | Specialist slots provided by this building. Valid keys are the names of specialists (as defined in [Specialists.json](https://github.com/yairm210/Unciv/tree/master/android/assets/jsons/Civ%20V%20-%20Gods%20&%20Kings/Specialists.json)), valid values are Integers, the amount of slots provided for this specialist |
-| civilopediaText | List | Default empty | see [civilopediaText chapter](5-Miscellaneous-JSON-files.md#civilopedia-text) |
+This file contains all the buildings and wonders you want to use in your mod.
+
+Each building has the following structure:
+
+| Attribute | Type | Default | Notes |
+| --------- | ---- | ------- | ----- |
+| name | String | Required | |
+| cost | Integer | 0 | Amount of production required to build the building |
```

Comment:
Technically false. Default is -1. Will give a warning if it's not unbuildable and don't have a specified tech to assume the cost from

Probably should be mentioned somewhere that default building costs are taken from the tech if unchanged (or -1)

Reply:
https://github.com/yairm210/Unciv/blob/93224f149135ba47f043ce73a3134fcaa1c4010c/core/src/com/unciv/models/ruleset/validation/RulesetValidator.kt#L588

Can you explain how does having a requiredTech helps? Will the building be instantly built?